packages/sdk/src/addresses/index.ts

```diff
@@ -0,0 +1,23 @@
+import {
+  ADDRESS_TYPE_TO_FORMAT,
+  NETWORK_TO_ADDRESS_TYPE_TO_REGEX,
+} from "./constants";
+import type { AddressFormat, AddressType } from "./types";
+import type { Network } from "../networks/types";
+
+export function getAddressFormat(
+  address: string,
+  network: Network,
+): AddressFormat {
+  const addressTypeToRegex = NETWORK_TO_ADDRESS_TYPE_TO_REGEX[network];
+  const addressTypes = Object.keys(addressTypeToRegex) as AddressType[];
+
+  // findLast because taproot addresses are being marked as segwit.
```

Comment:
This behaviour is identical to the previous code. (we don't return early so it returns the last entry being found).